src/users/users.service.ts

```diff
@@ -0,0 +1,105 @@
+import {
+  BadRequestException,
+  Injectable,
+  NotFoundException,
+} from '@nestjs/common';
+import { InjectModel } from '@nestjs/sequelize';
+import { ApiOperation, ApiResponse } from '@nestjs/swagger';
+import { Op } from 'sequelize';
+
+import { CreateUserDto } from './dto/create-user.dto';
+import { UpdateUserDto } from './dto/update-user.dto';
+import { SearchUserDto } from './dto/search-user.dto';
+import { User } from './users.model';
+
+@Injectable()
+export class UsersService {
+  constructor(@InjectModel(User) private userRepository: typeof User) {}
+
+  @ApiOperation({ summary: 'User creation' })
+  @ApiResponse({ status: 200, type: User })
+  async createUser(createUserDto: CreateUserDto) {
+    const user = await this.findByLogin(createUserDto.login);
+    if (user) {
+      throw new BadRequestException(
+        `User ${createUserDto.login} already exists in the database`,
+      );
+    }
+    return await this.userRepository.create(createUserDto);
+  }
+
+  @ApiOperation({ summary: 'Get all users' })
+  @ApiResponse({ status: 200, type: [User] })
+  async findAll(params: SearchUserDto) {
+    const substring =
+      params.loginSubstring === undefined ? '' : params.loginSubstring;
+    const limit = Number(params.limit);
+    const arrUsers = await this.userRepository.findAll({
+      where: {
+        isDeleted: false,
+        login: {
+          [Op.iLike]: `%${substring}%`,
+        },
+      },
+      order: ['login'],
+    });
```

Comment:
It seems to me details to work with BD should be in DAO layer

Reply:
done